framework/src/st_things/things_stack/easy-setup/easysetup_manager.c

```diff
@@ -277,14 +273,13 @@ esm_result_e esm_init_easysetup(int restart_flag, things_server_builder_s *serve
 		return ESM_ERROR;
 	}
 
-	if (server_builder != NULL) {
-		g_server_builder = server_builder;
-	}
-
 	if (g_server_builder == NULL) {
 		THINGS_LOG_V_ERROR(THINGS_ERROR, TAG, "[Error] g_server_builder is NULL.");
 		return ESM_ERROR;
+	} else {
+		g_server_builder = server_builder;
```

Comment:
Why did this logic move?
In this line, g_server_builder will be null if server_builder  was null.